### DISCUSSION TITLE
RLS from table return empty array when select with references table

### DISCUSSION THREAD

**Discussion by eposha** [supabase/supabase#19350] [Questions] [answered]
My RLS policy is 

```
BEGIN;
  ALTER POLICY "select by user_id" ON "public"."favorite_content" USING ((auth.uid() = user_id));
COMMIT;
```

when I use 
```
  const { data } = await supabase
    .from('favorite_content')
    .select('content_id')
    .in('content_id', ids);
```

I receive records as expected (rows where column user_id = auth.uid())

But whet I use 

```
    const { data, error } = await supabase
      .from('content')
      .select('*, favorite_content!inner(*)')
      .eq('favorite_content.user_id', userId);
```
or 

  ```
  const { data, error } = await supabase
      .from('content')
      .select('*, favorite_content(*)')
      .eq('favorite_content.user_id', userId);
```

or 

```
  const { data, error } = await supabase
      .from('content')
      .select('*, favorite_content!inner(*)')
```

I get empty data

but if I disable RLS and make query 

```
  const { data, error } = await supabase
      .from('content')
      .select('*, favorite_content!inner(*)')
      .eq('favorite_content.user_id', userId);
```

I receive data as expected

What is the problem with RLS or maybe another one?

**Comment by GaryAustin1** [comment]
You don't show your RLS on content.  You have to meet that obviously.
You are also not clear what tables you are disabling RLS on.
Also if you are using next.js caching it could be playing tricks on you.  But you don't say your environment.  
Just in case you are using next.js https://github.com/orgs/supabase/discussions/19084

**Reply by GaryAustin1** [reply, marked as answer]
Everything except your first query points to you don't have a signed in user like you think.
Every other query is like you don't have a session when you make the call as even just authenticated got blocked.
So if your first query was cached or somehow things changed after you did that test then it is the outlier.

There is not likely a bug in RLS or foreign keys.   Many, many users would be having issue.
So it is something in your code or environment.   
Just based on what you posted your calls and RLS look correct.  So seems to lean towards no user session when you make the call IMO.

**Reply by eposha** [reply]
It are delete and insert policy for favorite_content table and it is work like expected

<img width="307" alt="image" src="https://github.com/supabase/supabase/assets/54452472/869476d4-f52e-43da-96ca-aaacb41f32d4">

<img width="1119" alt="image" src="https://github.com/supabase/supabase/assets/54452472/52571642-15c0-4334-8954-654fd260e563">

**Reply by eposha** [reply]
@GaryAustin1 You are right, I didn't add cookies to createClient, my bad. I use one component for dynamic and for static pages (Next.js)

**Comment by eposha** [comment]
Table favorite_content

<img width="1314" alt="image" src="https://github.com/supabase/supabase/assets/54452472/5b6fce90-a9ca-4bea-a621-23e94efe4e56">

Table favorite_content
<img width="1214" alt="image" src="https://github.com/supabase/supabase/assets/54452472/09a82059-5a10-47dc-9682-f5fd0043a1f7">


Table content
<img width="1214" alt="image" src="https://github.com/supabase/supabase/assets/54452472/27f5b823-16c2-4cf1-9f21-42eac0e3b8ac">

**Comment by eposha** [comment]
I just edit RSL for favorite_table to this one and I still receive empty array
<img width="547" alt="image" src="https://github.com/supabase/supabase/assets/54452472/6c1eae81-0960-41c7-8583-02cf380e9c69">

Looks like query with reference table doesn't get auth data
